### PR TITLE
Install pnpm version 7 to support Node.js 14  in CI

### LINKED ISSUE #4643: Install specific versions of pnpm for CI
https://github.com/fastify/fastify/blob/f1991cf60a8034161ea226344c902903d7a6d138/.github/workflows/package-manager-ci.yml#L31-L34

https://github.com/fastify/fastify/actions/runs/4546742715/jobs/8015757124#step:4:16

image

### PR DESCRIPTION
**Pull request by Yvad60**
#### Description
PR fixes #4643 by installing pnpm version 7 which supports Node.js 14, using [setup pnpm action](https://github.com/marketplace/actions/setup-pnpm)

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

**Review by Eomm — APPROVED** [review]
we run this pipeline on push into `main`, but LGTM https://github.com/pnpm/action-setup

File: .github/workflows/package-manager-ci.yml [hunk 1]
```diff
@@ -29,9 +29,9 @@ jobs:
           node-version: ${{ matrix.node-version }}
 
       - name: Install with pnpm
-        run: |
-          curl -L https://unpkg.com/@pnpm/self-installer | node
-          pnpm install
+        uses: pnpm/action-setup@v2
+        with:
+          version 7
```
Comment (climba03003) [inline .github/workflows/package-manager-ci.yml:34]:
```suggestion
          version: 7
```

**Review by climba03003 — CHANGES_REQUESTED** [review]

File: .github/workflows/package-manager-ci.yml (hunk 1 above)
Comment (Yvad60) [reply .github/workflows/package-manager-ci.yml]: Ahh! thanks for the catch 👍

File: .github/workflows/package-manager-ci.yml [hunk 2]
```diff
@@ -29,9 +29,9 @@ jobs:
           node-version: ${{ matrix.node-version }}
 
       - name: Install with pnpm
-        run: |
-          curl -L https://unpkg.com/@pnpm/self-installer | node
-          pnpm install
+        uses: pnpm/action-setup@v2
+        with:
+          version: 7
```
Comment (jsumners) [inline .github/workflows/package-manager-ci.yml:34]:
```suggestion
        with:
          # pnpm@8 does not support Node.js 14
          version: 7
```

Comment (Yvad60) [reply .github/workflows/package-manager-ci.yml]: Comment added

**Review by climba03003 — CHANGES_REQUESTED** [review]
From the log, it still using `pnpm@8`.
Please also change

https://github.com/fastify/fastify/blob/bf1214879d9ad7baecba0b9af7d1e9ddbf41044e/.github/workflows/integration.yml#L34-L36

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**